include/swift/AST/DiagnosticsSema.def

```diff
@@ -2436,6 +2436,11 @@ ERROR(differentiable_attr_ambiguous_function_identifier,none,
 ERROR(differentiable_attr_forward_mode_unsupported,none,
       "forward-mode automatic differentiation is not supported yet", ())
 
+ERROR(compiler_evaluable_bad_context,none,
+      "@compilerEvaluable functions not allowed here", ())
```

Comment:
Perhaps change this diagnostic to indicate what contexts are valid for `@compilerEvaluable` functions?
E.g. `"@compilerEvaluable attribute is valid only for top-level functions and struct methods"`.

Reply:
I will improve it later (todo in TypeCheckAttr.cpp), with something like that and with information about the context.